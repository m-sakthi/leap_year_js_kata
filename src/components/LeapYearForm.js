import React, { Component } from 'react';

export default class LeapYearForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      year: '',
      isLeap: false
    }
  }

  setEnteredYear = (e) => {
    this.setState({ year: e.target.value });
  }

  validateLeapYear = () => {
    const { year } = this.state;
  }

  render() {

    return (
      <div data-testid="leapYearForm">

        <label>Enter year:</label>
        <input
          data-testid="leapYearFormEnterYear"
          type="number"
          onChange={this.setEnteredYear} />
        <br />
        <input
          data-testid="leapYearFormSubmit"
          type="button"
          value="validate"
          onClick={this.validateLeapYear}
        />
        <br />
        <span data-testid="leapYearResult">
          {`Entered year ${this.state.year}`}
        </span>

      </div>
    )
  }
}
