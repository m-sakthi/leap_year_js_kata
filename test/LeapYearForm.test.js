import React from 'react'; import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import LeapYearForm from '../src/components/LeapYearForm';

configure({ adapter: new Adapter() });

describe('LeapYearForm', () => {
  const yearInputEleQuery = 'input[data-testid="leapYearFormEnterYear"]';
  const divElemQuery = 'div[data-testid="leapYearForm"]';
  const validateBtnQuery = 'input[data-testid="leapYearFormSubmit"]';
  const resultSpanQuery = 'span[data-testid="leapYearResult"]';

  describe("LeapYearForm component render", () => {
    let component;

    beforeEach(() => {
      component = shallow(<LeapYearForm />);
    });

    it("should render properly", () => {
      expect(component).toMatchSnapshot();
    });

    it("should have a div element", () => {
      expect((component.find(divElemQuery)).length).toBe(1);
    });

    it("should have a label element", () => {
      expect((component.find('label')).length).toBe(1);
    });

    it("should have a input number element", () => {
      expect((component.find(yearInputEleQuery)).length).toBe(1);
    });

    it("should have a input button", () => {
      expect((component.find(validateBtnQuery)).length).toBe(1);
    });
  });

  describe("LeapYearForm component functions", () => {
    let component, yearInputEle, submitBtn;

    beforeEach(() => {
      component = shallow(<LeapYearForm />);
      yearInputEle = component.find(yearInputEleQuery);
      submitBtn = component.find(validateBtnQuery);
    });

    it("should set the entered year", () => {
      const year = 2020;

      yearInputEle.prop('onChange')({ target: { value: year } });

      expect(component.find(resultSpanQuery).text()).toEqual(`Entered year 2020`);
    })

  });

});