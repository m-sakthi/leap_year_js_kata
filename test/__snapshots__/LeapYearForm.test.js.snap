// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeapYearForm LeapYearForm component render should render properly 1`] = `ShallowWrapper {}`;
